resources/datasets/hmda/generate_msamd_csv.sh

```diff
@@ -0,0 +1,192 @@
+#     sed (Mac users: install a newer version of sed: `brew install gnu-sed`)
+#
+
+CSV_APPEND_FILE=
+CODE_COLUMN=1
+NAME_COLUMN=2
+INPUT_TYPE=csv
+PREPEND_YEAR=
+
+OPTIONS_GUIDE="Usage: convert-msalsx-to-html.sh [-c code-column] [-n name-column] [-a csv-append-file] [-t input-type] [-y year] input-file
+    -c	Default: $CODE_COLUMN
+    -n	Default: $NAME_COLUMN
+    -a	Append results to provided filename
+    -t  Options: csv, xlsx	Default: $INPUT_TYPE
+    -y  Prepend year as first column of csv
+"
+
+# Pull out the command line arguments (if any)
+while getopts ":c:n:a:t:y:h" opt; do
+    case $opt in
+        c)
+            CODE_COLUMN=$OPTARG
+            ;;
+        n)
+            NAME_COLUMN=$OPTARG
+            ;;
+        a)
+            CSV_APPEND_FILE=$OPTARG
+            ;;
+        t)
+            case "$OPTARG" in
+                csv|xlsx)
+                    INPUT_TYPE=$OPTARG
+                    ;;
+                *)
+                    echo "Invalid file type specified" >&2
+                    echo "$OPTIONS_GUIDE" >&2
+                    echo 1
+                    ;;
+            esac
+            ;;
+        y)
+            PREPEND_YEAR=$OPTARG
+            ;;
+        h)
+            echo "$OPTIONS_GUIDE" >&2
+            exit 0
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            echo "$OPTIONS_GUIDE" >&2
+            exit 1
+            ;;
+    esac
+done
+
+shift $((OPTIND-1))  #This tells getopts to move on to the next argument.
+
+if [ $# -ne 1 ]; then
+     echo "Please include the input file"
+     echo "$OPTIONS_GUIDE" >&2
+     exit 1
+fi
+
+if [[ -n "$CSV_APPEND_FILE" && -z "$PREPEND_YEAR" ]] ; then
+    echo "If appending to an existing CSV file, please define the year"
+    exit 1
+fi
+
+echo $INPUT_TYPE
+if [ $INPUT_TYPE = 'xlsx' ]; then
+    # Use xlsx2csv to convert the xlsx file: https://github.com/dilshod/xlsx2csv
+    # If xlsx2csv is not installed, exit
+    command -v xlsx2csv 2> /dev/null || {
+        echo >&2 "'xlsx2csv' is required for this script to execute. To install, run:
+pip install xlsx2csv 
+Note that root access may be required to run pip commands. Aborting.";
+        exit 1;
+    }
+fi
+
+# NOTE - if using OSX, you'll need to install a newer version of sed to support these queries.
+# `brew install gnu-sed` will create the gsed executable
+sed_bin='sed'
+command -v gsed 2> /dev/null && {
+  sed_bin='gsed'
+}
+
+sed_tweaks='
+  # sed input format       47894,"WASHINGTON-ARLINGTON-ALEXANDRIA, DC-VA-MD-WV",
+  # sed output format      47894,"Washington, Arlington, Alexandria - DC, VA, MD, WV"
+  # Note: also works if | is the delimiter
+
+  # remove any line that does not contain a 5 character integer (MSA code)
+  # i.e. remove any header rows
+  /[0-9]\{5\}[,|]/!d;
+
+  # remove the delimiter (| or ,) at the end of every line
+  s/[,|]$//;
+
+  # replace comma separating city block and state block
+  s/, / - /;
+
```

Comment:
ooh @m3brown i misread the purpose of the 2nd line. its replacing the commas & dashes with semi-colons. alright maybe we should add that to the comment as well